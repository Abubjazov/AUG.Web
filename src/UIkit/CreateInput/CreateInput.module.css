.root {
  display: flex;
  flex-direction: column;
  text-align: left;
  transition: all var(--transition-delay) linear;
  animation: var(--animation-visible);
}

.menu-closed {
  display: none;
}

.title {
  font-size: 15px;
  font-weight: 500;
  line-height: 15px;
  color: var(--light-textgray);
}

.button {
  width: 80px;
  height: 40px;
}

.input-wrapper {
  display: flex;
  justify-content: space-between;
  margin-top: 10px;
}

.root input {
  max-width: 190px;
}

@media (width <= 330px) {
  .root input {
    width: 150px;
  }
}
