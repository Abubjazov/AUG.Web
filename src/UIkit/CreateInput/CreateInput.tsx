import { FC } from 'react'

import { useInput } from 'hooks/useInput/useInput'
import BaseButton from 'uikit/BaseButton'
import { BaseButtonMode } from 'uikit/BaseButton/BaseButton'
import BaseInput from 'uikit/BaseInput'
import { combineClasses as cc } from 'utils/combineClasses/combineClasses'

import styles from './CreateInput.module.css'

export interface CreateInputProps {
  userStyles?: string
  title: string
  placeholder: string
  menuOpened: boolean
  loading?: boolean
  inputValidators?: object
  onClick?: (inputText: string) => void
}

const CreateInput: FC<CreateInputProps> = ({
  userStyles = '',
  inputValidators,
  menuOpened,
  loading,
  placeholder,
  title,
  onClick,
}) => {
  const value = useInput('', { ...inputValidators })

  const submitHandler = () => {
    value.onBlur()

    if (!value.errors.length && onClick) {
      onClick(value.value as string)

      value.onChange({ target: { value: '' } })
      value.setIsDirty(false)
    }
  }

  return (
    <div
      className={cc([
        styles.root,
        menuOpened ? '' : styles['menu-closed'],
        userStyles,
      ])}
      onBlur={() => value.setIsDirty(false)}
    >
      <span className={styles.title}>{title}</span>

      <div className={styles['input-wrapper']}>
        <BaseInput
          dataTestId={'tag-name-input'}
          type={'text'}
          name={'tag-name-input'}
          placeholder={placeholder}
          value={value.value}
          onChange={value.onChange}
          onBlur={value.onBlur}
          errors={value.errors}
          isDirty={value.isDirty}
        />

        <BaseButton
          userStyles={styles.button}
          label={'Create'}
          mode={BaseButtonMode.CONTAINED_BLUE}
          onClick={submitHandler}
          loading={loading}
        />
      </div>
    </div>
  )
}

export default CreateInput
