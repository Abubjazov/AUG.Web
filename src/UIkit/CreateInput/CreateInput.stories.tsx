import type { Meta, StoryObj } from '@storybook/react'

import CreateInput, { CreateInputProps } from './CreateInput'

import '/src/index.css'

const meta: Meta<CreateInputProps> = {
  component: CreateInput,
  title: 'UIkit/Create Input',
  parameters: {
    docs: {
      description: {
        component: 'Standard interface component - basic input group',
      },
    },
  },
  tags: ['autodocs'],
  argTypes: {
    userStyles: {
      description: 'Passing additional custom styles for root element',
    },
    title: { description: 'Input group title' },
    placeholder: { description: 'Input placeholder' },
    menuOpened: { description: 'Hiding the input group' },
    loading: {
      description:
        'Appearance of the input group when loading content (spinner)',
    },
    inputValidators: {
      description: 'Optional validators list for input',
    },
    onClick: {
      description: 'Optional click handler',
    },
  },
}

export default meta

type Story = StoryObj<CreateInputProps>

export const Default: Story = {
  args: {
    menuOpened: true,
    title: 'Title',
    placeholder: 'placeholder',
    inputValidators: {
      isEmpty: { value: true, message: 'Input value required' },
      minLength: {
        value: 3,
        message: 'Minimum input value length 3',
      },
    },
  },
  decorators: [
    Story => (
      <div style={{ width: '290px' }}>
        <Story />
      </div>
    ),
  ],
}
