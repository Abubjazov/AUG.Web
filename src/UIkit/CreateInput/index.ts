import CreateInput from './CreateInput'

export default CreateInput
