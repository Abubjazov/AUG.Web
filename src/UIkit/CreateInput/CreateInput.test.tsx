import { fireEvent, screen, render } from '@testing-library/react'

import CreateInput from './CreateInput'

describe('CreateInput', () => {
  test('should render CreateInput default', () => {
    const { asFragment } = render(
      <CreateInput
        title={'Title'}
        placeholder={'Placeholder'}
        menuOpened={true}
      />,
    )

    expect(asFragment()).toMatchSnapshot()
  })

  test('should render CreateInput after clicking Create', () => {
    const { asFragment } = render(
      <CreateInput
        title={'Title'}
        placeholder={'Placeholder'}
        menuOpened={true}
      />,
    )

    fireEvent.change(screen.getByTestId('create-input'), {
      target: { value: 'New tag' },
    })

    expect(asFragment()).toMatchSnapshot()

    fireEvent.click(screen.getByTestId('base-button'))

    expect(asFragment()).toMatchSnapshot()
  })
})
