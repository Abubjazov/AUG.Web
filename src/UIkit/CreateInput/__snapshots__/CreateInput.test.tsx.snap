// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CreateInput > should render CreateInput after clicking Create 1`] = `
<DocumentFragment>
  <div
    class="_root_b62a2a  "
  >
    <span
      class="_title_b62a2a"
    >
      Title
    </span>
    <div
      class="_input-wrapper_b62a2a"
    >
      <div
        class="_root_672eb8"
      >
        <input
          data-testid="tag-name-input"
          name="tag-name-input"
          placeholder="Placeholder"
          type="text"
          value="New tag"
        />
      </div>
      <button
        class="_root_91d6ec _contained-blue_91d6ec  _button_b62a2a"
        data-testid="base-button"
        type="button"
      >
        Create
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`CreateInput > should render CreateInput after clicking Create 2`] = `
<DocumentFragment>
  <div
    class="_root_b62a2a  "
  >
    <span
      class="_title_b62a2a"
    >
      Title
    </span>
    <div
      class="_input-wrapper_b62a2a"
    >
      <div
        class="_root_672eb8"
      >
        <input
          data-testid="tag-name-input"
          name="tag-name-input"
          placeholder="Placeholder"
          type="text"
          value=""
        />
      </div>
      <button
        class="_root_91d6ec _contained-blue_91d6ec  _button_b62a2a"
        data-testid="base-button"
        type="button"
      >
        Create
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`CreateInput > should render CreateInput default 1`] = `
<DocumentFragment>
  <div
    class="_root_b62a2a  "
  >
    <span
      class="_title_b62a2a"
    >
      Title
    </span>
    <div
      class="_input-wrapper_b62a2a"
    >
      <div
        class="_root_672eb8"
      >
        <input
          data-testid="tag-name-input"
          name="tag-name-input"
          placeholder="Placeholder"
          type="text"
          value=""
        />
      </div>
      <button
        class="_root_91d6ec _contained-blue_91d6ec  _button_b62a2a"
        data-testid="base-button"
        type="button"
      >
        Create
      </button>
    </div>
  </div>
</DocumentFragment>
`;
